gui/system/forms.py

```diff
@@ -2510,9 +2510,22 @@ class CertificateAuthorityCreateInternalForm(MiddlewareModelForm, ModelForm):
         required=True,
         help_text=models.CertificateAuthority._meta.get_field('cert_name').help_text
     )
+    cert_key_type = forms.ChoiceField(
+        label=_('Key Type'),
+        help_text=_('Please select a key type to proceed with'),
+        required=True,
+        choices=()
+    )
+    cert_ec_curve = forms.ChoiceField(
+        label=_('EC Key Curve'),
+        help_text=_('Please select a curve to use to generate EC key'),
```

Comment:
As above, the help text should help the user pick the best choice.

Reply:
There are different strengths/weaknesses for each curve, here is a site which provides some metrics for performance - https://tls.mbed.org/kb/cryptography/elliptic-curve-performance-nist-vs-brainpool. Please let me know how you think we should move forward with this one. Rest of the forms use same fields, so these two fields can be copied over there then. Also if you would like me to more information, I can do that but we should be perhaps precise. There are lots of curves for EC, we only support 4 right now. We can add more later as users request. To summarise the current choices, `brainpool*` curves can be considered more secure in some scenarios and `secp*` ones can be considered faster. Both should be good, users with specific use cases might go for different ones.